src/main/java/org/opensearch/flowframework/indices/FlowFrameworkIndicesHandler.java

```diff
@@ -0,0 +1,176 @@
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.indices;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.admin.indices.create.CreateIndexRequest;
+import org.opensearch.action.admin.indices.create.CreateIndexResponse;
+import org.opensearch.action.admin.indices.mapping.put.PutMappingRequest;
+import org.opensearch.action.admin.indices.settings.put.UpdateSettingsRequest;
+import org.opensearch.client.Client;
+import org.opensearch.cluster.metadata.IndexMetadata;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.common.xcontent.XContentType;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.flowframework.exception.FlowFrameworkException;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static org.opensearch.flowframework.constant.CommonValue.META;
+import static org.opensearch.flowframework.constant.CommonValue.NO_SCHEMA_VERSION;
+import static org.opensearch.flowframework.constant.CommonValue.SCHEMA_VERSION_FIELD;
+
+/**
+ * Flow Framework Indices Handler
+ */
+public class FlowFrameworkIndicesHandler {
+
+    private static final Logger logger = LogManager.getLogger(FlowFrameworkIndicesHandler.class);
+    private static final Map<String, Object> indexSettings = Map.of("index.auto_expand_replicas", "0-1");
+    private static final Map<String, AtomicBoolean> indexMappingUpdated = new HashMap<>();
+
+    private ClusterService clusterService;
+    private Client client;
+
+    /**
+     * Handler constructor
+     * @param clusterService cluster service
+     * @param client client
+     */
+    public FlowFrameworkIndicesHandler(ClusterService clusterService, Client client) {
+        this.clusterService = clusterService;
+        this.client = client;
+    }
+
+    static {
+        for (FlowFrameworkIndex flowFrameworkIndex : FlowFrameworkIndex.values()) {
+            indexMappingUpdated.put(flowFrameworkIndex.getIndexName(), new AtomicBoolean(false));
+        }
+    }
+
+    /**
+     * Initiate global context index if it's absent
+     * @param listener action listner
+     */
+    public void initGlobalContextIndexIfAbsent(ActionListener<Boolean> listener) {
+        initFlowFrameworkIndexIfAbsent(FlowFrameworkIndex.GLOBAL_CONTEXT, listener);
+    }
+
+    /**
+     * General method for initiate flow framework indices or update index mapping if it's needed
+     * @param index flow framework index
+     * @param listener action listener
+     */
+    public void initFlowFrameworkIndexIfAbsent(FlowFrameworkIndex index, ActionListener<Boolean> listener) {
+        String indexName = index.getIndexName();
+        String mapping = index.getMapping();
+
+        try (ThreadContext.StoredContext threadContext = client.threadPool().getThreadContext().stashContext()) {
```

Comment:
From what I understand about security, we only need to stash the thread context if this particular index is managed by the security plugin's list of protected indices. This allows the plugin to do CRUD operations on this protected index. Is the plan to add all flow framework indices to the security plugin's list of protected indices, and if not, why stash the thread context?